code/modules/power/fission/core.dm

```diff
@@ -0,0 +1,302 @@
+								var/d_neutron_energy = max(0, interaction_units * (mat.fission_energy - old_neutron_energy)/HI_NEUT_DIVISOR)
+								last_neutron_energy_increase += d_neutron_energy
+								neutron_energy += d_neutron_energy
+								neutron_flux += d_neutron_flux
+								last_neutron_flux_increase += d_neutron_flux
+
+							if(mat.fission_heat)
+								air_contents.add_thermal_energy(removed*mat.fission_heat)
+
+					// A simplification of neutron capture and/or subsequent beta decay.
+					if(INTERACTION_ABSORPTION)
+						if(length(mat.absorption_products))
+							if(fr.matter[mat_type] - interaction_units < 0)
+								removed = fr.matter[mat_type]
+								fr.matter -= mat_type
+							else
+								removed = interaction_units
+								fr.matter[mat_type] -= removed
+
+							for(var/byproduct_type in mat.absorption_products)
+								fr.matter[byproduct_type] += removed * mat.absorption_products[byproduct_type]
+						
+						var/d_neutron_flux = mat.neutron_absorption*removed
+						neutron_flux -= d_neutron_flux
+						last_neutron_flux_decrease += d_neutron_flux
+
+					// Neutron moderation, reduction of neutron energy towards the target value for that material.
+					if(INTERACTION_SCATTER)
+						// Neutron moderators can only slow, not speed up neutrons. Check against the current neutron energy to ensure this.
+						if(mat.moderation_target > neutron_energy)
+							continue
+						var/d_neutron_energy = min(neutron_energy, interaction_units * (old_neutron_energy - mat.moderation_target)/LO_NEUT_DIVISOR)
+						last_neutron_energy_decrease += d_neutron_energy
+						neutron_energy = max(mat.moderation_target, neutron_energy - d_neutron_energy)
+
+	// Neutron energy must pass a threshold for there to be danger of radiation.
+	if(neutron_energy >= MIN_RADIATION_ENERGY)
+		var/radiation_power = Clamp(old_neutron_energy/1000 * old_neutron_flux/1000, 0, 40)
+		SSradiation.radiate(src, round(radiation_power))
+
+	// Control rods reduces neutron flux by 1/5th at full depth.
+	last_neutron_flux_decrease += round(neutron_flux*(control_rod_depth/CONTROL_ROD_DIVISOR))
+	neutron_flux = round(neutron_flux*(1 - (control_rod_depth/CONTROL_ROD_DIVISOR)))
+
+	if(!neutron_flux)
+		neutron_energy = 0
+
+	net_neutron_energy_change = last_neutron_energy_increase - last_neutron_energy_decrease
+	net_neutron_flux_change = last_neutron_flux_increase - last_neutron_flux_decrease
+
+/obj/machinery/atmospherics/unary/fission_core/proc/meltdown()
+	if(melted_down)
+		return
+	melted_down = TRUE
+	var/total_radioactivity = 0
+
+	for(var/obj/item/fuel_assembly/rod in fuel_rods)
+		if(total_radioactivity >= MAX_RADS)
+			break
+		for(var/mat_type in rod.matter)
+			var/decl/material/mat = decls_repository.get_decl(mat_type)
+			total_radioactivity += mat.radioactivity*SHEET_MATERIAL_AMOUNT / rod.matter[mat_type]
+
+	visible_message(SPAN_DANGER("\The [src] explodes, blowing out its radioactive contents!"))
```

Comment:
Could a reactor with no fuel rods conceivably melt down and produce no radioactivity?